File: 프로그래머스/unrated/134240. 푸드 파이트 대회/푸드 파이트 대회.cs

using System;
using System.Linq;

public class Solution {
    public string solution(int[] food) {
        string answer = "";
        for(int i = 1 ;i < food.Length ; i ++  )
        {
         if(food[i]%2 ==1) 
         {
          food[i] -=1;   
         }
            int count = food[i]/=2;
            for(int j = 0; j <count; j++ ){
            answer += i.ToString();//food[i].ToString();
            }
            
        }
        
            
        
        return answer + "0" + new string(answer.Reverse().ToArray());
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;

string solution(vector<int> food) {
    vector<int>num;
    string answer = "";
    for(int i = 1 ; i < food.size(); i++)
    {
        for(int j = 0; j < food[i]/2; j ++)
        {
        num.push_back(i);
        answer += to_string(i);
        }
    }
    answer += "0";
    
    for(int i = num.size()-1; i >= 0 ; i--)
    {
        answer += to_string(num[i]);
    }
    return answer;
}